### ISSUE TITLE
find_home() causing "path name too long" error in Windows

### ISSUE THREAD

**Issue by KricketK** [dmulholland/ivy#11]
Error Text: 
```
ivy.main()
 Traceback (most recent call last):
   File "<stdin>", line 1, in <module>
  File "C:\Python\Python36\lib\site-packages\ivy\__init__.py", line 55, in main
    site.init()
  File "C:\Python\Python36\lib\site-packages\ivy\site.py", line 33, in init
    load_site_config()
  File "C:\Python\Python36\lib\site-packages\ivy\site.py", line 45, in load_site_config
    if home() and isfile(home('config.py')):
  File "C:\Python\Python36\lib\site-packages\ivy\site.py", line 79, in home
    path = cache.get('home') or cache.setdefault('home', find_home())
  File "C:\Python\Python36\lib\site-packages\ivy\site.py", line 66, in find_home
    if isfile(join(path, 'config.py')):
  File "C:\Python\Python36\lib\genericpath.py", line 30, in isfile
    st = os.stat(path)
ValueError: stat: path too long for Windows
```

![original error](https://user-images.githubusercontent.com/13955147/36051362-b5da0ee4-0d9e-11e8-9e4b-c07885076c5d.PNG)
![image of path created by ivy](https://user-images.githubusercontent.com/13955147/36051360-b59af86c-0d9e-11e8-8ecd-77b8d0e3c8f7.jpg)
![culprit line](https://user-images.githubusercontent.com/13955147/36051361-b5b1c4c0-0d9e-11e8-8f0a-d83ef9edfe73.jpg)

When I executed the code and implemented a PyCharm debugger, it revealed that ivy was trying to analyze C:\\the\\path\\here\\..\\..\\..\\..\\..\\..\\ on and on, ad infinitum, causing the error.

I traced back the error path to the find_home function in ivy\site.py. I am unsure what the initializaiton is supposed to output as I could not get it to work without modifications, but my interpretation of the notes suggested that the config.py file is supposed to end up in the site home directory. So I wrote the following code to make that config.py file within the working directory should the while loop in find_home() fail to find home in the cwd and parent directories.

    def find_home():

        # I'm importing a few things for what I'm going to do
        from os.path import split
        from shutil import copyfile
    
        # path remains the same, but I'm adding ivyconfigpath to locate the original config.py file. I'm sure
        # I'm sure there's a more elegant way to do this, particularly with your cache system and such, but I
        # wanted to keep things simple for myself without fully knowing the ins and outs of Ivy
        ivyconfigpath = join(os.path.dirname(os.path.dirname(__file__)), "ivy/ini/config.py")
        path = os.getcwd()
        # workingpath is what I can modify in the while loop search without actually affecting path itself
        workingpath = path
        
        # I changed the condition of your while statement. I felt that it was likely path would be a directory
        # as we just defined it using os.getcwd()
        # but what I needed to prevent is recursion while looping through folder. When split(path)[1] == ''
        # we know that means there is no second directory and we have reached the root drive
        # so I could avoid messing with any dependencies, at all outcomes, the workingpath used in the
        # while loop is the value set to path.
        while split(workingpath)[-1] != '':
            if isfile(join(workingpath, 'config.py')):
                path = workingpath
                return os.path.abspath(path)
            elif isfile(join(path, 'site.py')):
                path = workingpath
                return os.path.abspath(path)
            elif isdir(join(path, 'src')) and isdir(join(path, 'out')):
                path = workingpath
                return os.path.abspath(path)
            else:
                # removing the base of the path with each loop so we can search through parent directories
                # I may be entirely wrong, but I suspected that was the purpose of the join(path, '..')
                # and that it was supposed to act like $ cd ..
                workingpath = split(workingpath)[0]

        # here I copy the config.py folder from the ivy directory to the current working directory
        # this means that upon ivy init, the current working directory becomes the site's home directory.
        # again, I don't know if that was your purpose, but it worked for my purposes and allowed proper
        # execution of the ivy module.
        newconfigfile = "config.py"
        copyfile(ivyconfigpath, join(path, newconfigfile))
        return ''

I hope that this may be helpful to anyone else who finds Ivy is encountering a weird file path length error at the very least and apologies if it is less elegant than it could be.

Please offer corrections if there is a possibility I did not account for that is glaringly obvious to someone else!

Thank you for all your hard work!

**Comment by dmulholland** [comment]
Thanks for the detailed bug report, that was very helpful.

I've just pushed a commit to master here on Github which will hopefully fix the issue. I don't have a Windows system handy to test it on so if you can give it a try and let me know if it works for you I'd really appreciate it.

If you download the updated version from Github you should be able to run it directly without installing it just by passing Python the path to the package directory:

    $ python path/to/ivy/directory

If it works it should print the help text. You can then append any commands you want to run, e.g.

    $ python path/to/ivy/directory init

**Comment by rawrgulmuffins** [comment]
@KricketK Also ran into a issue with `ivy tree`. I'm writing this comment so they'll get a notification and write a bug report about that issue as well.

**Comment by dmulholland** [comment]
Version 1.1.0 should fix the original issue, 1.1.1 should fix the unicode printing problem. Thanks for the report.
